astropy/units/quantity_helper/converters.py

```diff
@@ -92,16 +99,17 @@ def __setitem__(self, key, value):
         # mean that something is not implemented, but this means an
         # extra if clause for the output, slowing down the common
         # path where a ufunc is supported.
-        if value is None:
-            self.UNSUPPORTED |= {key}
-            self.pop(key, None)
-        else:
-            super().__setitem__(key, value)
-            self.UNSUPPORTED -= {key}
+        with self._lock:
+            if value is None:
+                self.unsupported |= {key}
+                self.pop(key, None)
+            else:
+                super().__setitem__(key, value)
+                self.unsupported -= {key}
 
 
 UFUNC_HELPERS = UfuncHelpers()
-UNSUPPORTED_UFUNCS = UFUNC_HELPERS.UNSUPPORTED
+UNSUPPORTED_UFUNCS = UFUNC_HELPERS.unsupported
```

Comment:
Will replacing `UNSUPPORTED` with lowercase version break people's code? Do people use `UNSUPPORTED` from this class directly? Perhaps it is a smaller price to pay to break some PEP 8 and do a `self.UNSUPPORTED |= ...` and such. 💭

Reply:
You're right, I'd missed that users could directly access it as UFUNC_HELPERS.UNSUPPORTED or UfuncHelpers.UNSUPPORTED rather than through this alias, even though all the astropy code uses the alias. I'll change it back to the way it was before (uppercase, class attribute) and think up some way to isolate it in the test.

Reply:
Done.